src/main/java/no/entur/uttu/stopplace/filter/StopPlacesFilterer.java

```diff
@@ -0,0 +1,204 @@
+package no.entur.uttu.stopplace.filter;
+
+import static no.entur.uttu.error.codes.ErrorCodeEnumeration.INVALID_STOP_PLACE_FILTER;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import no.entur.uttu.error.codederror.CodedError;
+import no.entur.uttu.error.codedexception.CodedIllegalArgumentException;
+import org.rutebanken.netex.model.Quay;
+import org.rutebanken.netex.model.StopPlace;
+
+public class StopPlacesFilterer {
+
+  /**
+   * Filter the set of all stop places by various criteria;
+   * Filtering by quay id-s doesn't take into account any other filters that may be by chance provided along;
+   * Filtering by limiting the final number of stop places if applied at the very last step;
+   * All the other types of filtering one-by-one, in iteration, narrow down the stop places set;
+   * @param allStopPlaces
+   * @param stopPlaceByQuayRefIndex
+   * @param filters
+   * @return
+   */
+  public List<StopPlace> filter(
+    List<StopPlace> allStopPlaces,
+    Map<String, StopPlace> stopPlaceByQuayRefIndex,
+    List<StopPlaceFilter> filters
+  ) {
+    Optional<StopPlaceFilter> quayIdsFilterOpt = findFilterByClass(
+      filters,
+      QuayIdFilter.class
+    );
+
+    if (quayIdsFilterOpt.isPresent()) {
+      return getStopPlacesByQuayIds(
+        (QuayIdFilter) quayIdsFilterOpt.get(),
+        stopPlaceByQuayRefIndex
+      );
+    }
+
+    List<StopPlaceFilter> filtersToIterateThrough = filters
+      .stream()
+      .filter(StopPlaceFilter::isAppliedCompositely)
+      .toList();
+    List<StopPlace> filteredStopPlaces = allStopPlaces
+      .stream()
+      .filter(s -> isStopPlaceToBeIncluded(s, filtersToIterateThrough))
+      .toList();
+
+    Optional<StopPlaceFilter> limitFilterOpt = findFilterByClass(
+      filters,
+      LimitStopPlacesQuantityFilter.class
+    );
+
+    return limitFilterOpt
+      .map(stopPlaceFilter ->
+        limitNumberOfStopPlaces(
+          ((LimitStopPlacesQuantityFilter) stopPlaceFilter).limit(),
+          filteredStopPlaces
+        )
+      )
+      .orElse(filteredStopPlaces);
+  }
+
+  /**
+   * Find an instance of a certain kind of filter;
+   * Normally, there is only one occurrence of a certain kind of filter in the filters list
```

Comment:
This is ok but it should be documented on the SPI: https://github.com/entur/uttu/blob/0c81794329b189031f23ea8c2e7a491cb8c80cb7/src/main/java/no/entur/uttu/stopplace/spi/StopPlaceRegistry.java#L39